File: 백준/Bronze/2231. 분해합/분해합.py
n = int(input())
x = 0
result = 0

for i in range(1, n+1):
    for j in str(i) :
        x += int(j)
    x += i

    if (x != n) :
        x = 0
    else :
        result = i
        break
print(result)

File: 백준/Bronze/2231. 분해합/분해합.cc
#include <iostream>
using namespace std;

int solve(int n){
	int answer = 0;
	
	while(n>0){
		answer += n % 10;
		n /= 10;
	}
	
	return answer;
}

int main(){
	int N, result = 0;
	cin >> N;
	
	for (int i=1; i<N+1; i++){
		int x = i + solve(i);
		
		if(x == N){
			result = i;
			break;
		}
	}
	
	cout << result;
	
	return 0;
}